File: be/src/main/java/nl/zzpmatcher/profile/controller/ProfileProjection.java
package nl.zzpmatcher.profile.controller;

import nl.zzpmatcher.profile.business.Profile;
import nl.zzpmatcher.profile.business.Tag;

public class ProfileProjection {
    private final String username;
    private final String[] tags;

    private ProfileProjection(String username, String[] tags) {
        this.username = username;
        this.tags = tags;
    }

    public String getUsername() {
        return username;
    }

    public String[] getTags() {
        return tags;
    }

    static ProfileProjection of(Profile profile) {
        if(profile==null)
            return null;

        final String[] tags = profile.getTags().stream().map(Tag::getId).toArray(String[]::new);

        return new ProfileProjection(profile.getId(), tags);
    }
}

File: be/src/main/java/nl/zzpmatcher/profile/controller/ProfileProjection.kt
package nl.zzpmatcher.profile.controller

import nl.zzpmatcher.profile.business.Profile
import nl.zzpmatcher.profile.business.Tag
import kotlin.streams.toList

data class ProfileProjection private constructor(val username: String?, val tags: Array<String>) {
    companion object {

        internal fun of(profile: Profile): ProfileProjection {

            val tags= profile.tags!!.stream().map { it.id!! }.toList().toTypedArray()

            return ProfileProjection(profile.id, tags)
        }
    }
}

